quelpa.el

```diff
@@ -1933,18 +1933,18 @@ to install.
 
 When `quelpa' is called interactively with a prefix argument (e.g
 \\[universal-argument] \\[quelpa]) it will try to upgrade the
-given package even if the global var `quelpa-upgrade-p' is set to
-nil."
+given package and remove old versions.  Otherwise, the custom
+variables `quelpa-upgrade-p' and `quelpa-autoremove-p' control
+whether these actions should take place by default."
   (interactive (list nil))
   (run-hooks 'quelpa-before-hook)
   (when (quelpa-setup-p) ;if init fails we do nothing
     (let* ((arg (or arg
                     (let ((quelpa-melpa-recipe-stores
                            `(,@quelpa-melpa-recipe-stores ,quelpa-cache)))
                       (quelpa-interactive-candidate))))
-           (quelpa-upgrade-p (if current-prefix-arg t quelpa-upgrade-p)) ;shadow `quelpa-upgrade-p'
-           (quelpa-stable-p quelpa-stable-p) ;shadow `quelpa-stable-p'
```

Comment:
You shouldn't remove the `quelpa-stable-p` check here. It's for shadowing `quelpa-stable-p` so nested call will not modify the global `quelpa-stable-p`

Reply:
Thanks. I looked at the code again. I must say, this is some of the most egregious abuse of dynamic scoping I've ever seen :P